component.go

```diff
@@ -0,0 +1,36 @@
+package spec3
+
+import "github.com/go-openapi/spec"
+
+// Components holds a set of reusable objects for different aspects of the OAS.
+// All objects defined within the components object will have no effect on the API
+// unless they are explicitly referenced from properties outside the components object.
+//
+// see: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#componentsObject
+type Components struct {
```

Comment:
initially, this struct will hold only the minimal set of fields - only the ones that are necessary to convert the spec from [v2](https://github.com/kubernetes/kubernetes/blob/master/api/openapi-spec/swagger.json) to `v3`